### ISSUE TITLE
fsck.ocfs2/include/strings.h conflicts with system header from glibc master (2.27)

### ISSUE THREAD

**Issue by cheese** [markfasheh/ocfs2-tools#17]
Fedora rawhide(28)
glibc-2.26.90-14.fc28.x86_64
```
# cd fsck.ocfs2
# gcc -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -pipe    -I../include -Iinclude  -DVERSION=\"1.8.5\"  -MD -MP -MF ./.pass4.d -o pass4.o -c pass4.c
In file included from /usr/include/string.h:431:0,
                 from ../include/ocfs2/ocfs2.h:41,
                 from pass4.c:32:
include/strings.h:37:1: error: unknown type name ‘errcode_t’; did you mean ‘mode_t’?
 errcode_t o2fsck_strings_insert(o2fsck_strings *strings, char *string,
 ^~~~~~~~~
 mode_t
```

/usr/include/string.h includes <strings.h> from fsck.ocfs2/include/strings.h.

A fix may be renaming fsck.ocfs2/include/strings.h.

**Comment by cheese** [comment]
I just run 'make' and then retry compiling the failed source.

**Comment by ganghe** [comment]
Hello Robin,

Do you know the building error root cause? it was brought by glibc version upgrade?
Since in the past, there was no this build error.


Thanks
Gang

**Comment by cheese** [comment]
Yes, it is brought by glibc version update.

In new version of glibc, string.h includes strings.h. This is a change of glibc master.

The direct course is:
pass4.c indirectly includes glibc string.h. And glibc string.h would find strings.h and found it in fsck.ocfs2/include/strings.h instead of strings.h from glibc.
And <et/com_err.h> has not yet  been included, so 'errcode_t' is not defined.

As i suggested above, a fix may be renaming fsck.ocfs2/include/strings.h.

**Comment by ganghe** [comment]
Hello Robin,

Do you know which Linux distribution can make this building error happen? 


Thanks
Gang

**Comment by cheese** [comment]
I said in the initial report that I tested it on Fedora Rawhide(28)

**Comment by ganghe** [comment]
Ok, Thanks a lot, I also got this error in opensuse Tumbleweed build.
The fix was commited.

https://github.com/markfasheh/ocfs2-tools/commit/0ffd58b223e24779420130522ea8ee359505f493


Thanks
Gang

**Comment by panlinux** [comment]
Since the fix was committed, could someone perhaps close this issue?

**Comment by josephhz** [comment]
Closed as it was fixed by commit 0ffd58b223e2 from Gang.
